src/server/scripts/Kalimdor/zone_silithus.cpp

```diff
@@ -1214,41 +1218,88 @@ class go_wind_stone : public GameObjectScript
+                        ObjectVector primaryTarget;
+                        primaryTarget.push_back(player);
+                        sai->GetScript()->AddToStoredTargetList(primaryTarget, 1);
+
+                        // Add player's group (if applicable) within 50yds to SmartAI stored target list 2
+                        if (Group* group = player->GetGroup())
+                        {
+                            ObjectVector secondaryTargets;
+                            for (GroupReference* groupRef = group->GetFirstMember(); groupRef != nullptr; groupRef = groupRef->next())
+                                if (Player* member = groupRef->GetSource())
+                                    if (member != player && member->IsInMap(summons) && member->GetDistance(summons) < 50.0f)
+                                        secondaryTargets.push_back(member);
+
+                            if (secondaryTargets.size() > 0)
+                                sai->GetScript()->AddToStoredTargetList(secondaryTargets, 2);
+                        }
+
+                        switch (summons->GetEntry())
+                        {
                         case NPC_TEMPLAR_FIRE:
                         case NPC_TEMPLAR_WATER:
                         case NPC_TEMPLAR_AIR:
                         case NPC_TEMPLAR_EARTH:
-                            summons->AI()->Talk(SAY_TEMPLAR_AGGRO, player);
+                            sai->Talk(SAY_TEMPLAR_AGGRO, player);
                             break;
 
                         case NPC_DUKE_FIRE:
                         case NPC_DUKE_WATER:
                         case NPC_DUKE_EARTH:
                         case NPC_DUKE_AIR:
-                            summons->AI()->Talk(SAY_DUKE_AGGRO, player);
+                            sai->Talk(SAY_DUKE_AGGRO, player);
                             break;
                         case NPC_ROYAL_FIRE:
                         case NPC_ROYAL_AIR:
                         case NPC_ROYAL_EARTH:
                         case NPC_ROYAL_WATER:
-                            summons->AI()->Talk(YELL_ROYAL_AGGRO, player);
+                            sai->Talk(YELL_ROYAL_AGGRO, player);
                             break;
+                        }
+                        // Next actions happen in smart_scripts
+                        // On Summon: Set Root, Set Immune to PC, Create Timed Event 1 (5s), Create Timed Event 2 (30s)
+                        // On Timed Event 1 (5s): Unset Root, Unset Immune to PC, Add 100.0f Threat (Stored List 1), Add 50.0f Threat (Stored List 2)
+                        // On Reset: Set Counter 1 +1
+                        // On Counter 1 = 2: Despawn Self In 1000ms
+                        // On Timed Event 2 (30s): Despawn Self Instant
+                        // On Aggro: Remove Timed Event 2
+                        // In Combat: Use NPC specific spells
+                    }
+                    else {
```

Comment:
```suggestion
                    else
                    {
```
Applying TC codestyle